talpid-tunnel-config-client/proto/tunnel_config.proto

```diff
@@ -19,7 +7,69 @@ package tunnel_config;
 service PostQuantumSecure {
   // PskExchangeExperimentalV0 uses the common API defined by LibOQS.  See:
   // https://github.com/open-quantum-safe/liboqs
+  // This endpoint is deprecated in favor for `PskExchangeExperimentalV1`. Please use that instead.
   rpc PskExchangeExperimentalV0(PskRequestExperimentalV0) returns (PskResponseExperimentalV0) {}
+
+  // Allows deriving a preshared key (PSK) using one or multiple PQ-secure key-encapsulation
+  // mechanisms (KEM). The preshared key is added to WireGuard's preshared-key field in a new
+  // ephemeral peer (PQ-peer). This makes the tunnel resistant towards attacks using
+  // quantum computers.
+  //
+  // The VPN server associates the PQ-peer with the peer who performed the exchange. Any
+  // already existing PQ-peer for the normal peer is replaced. Each normal peer can have
+  // at most one PQ-peer.
+  //
+  // The PQ-peer is mutually exclusive to the normal peer. A handshake from the normal peer
+  // throws out the corresponding PQ-peer from WireGuard and vice versa.
```

Comment:
This could be clarified. When the PQ-peer first tries to handshake, the associated normal peer is unloaded from the WireGuard interface but remembered by `wg-manager`. If at a later time, the normal peer again tries to handshake, the PQ-peer is unloaded and remembered. A client can switch back and forth between the peers without having to renegotiate, for as long as the wg-manager instance stays alive.

Reply:
I'll see if I can find a good wording. I don't want to mention `wg-manager` as the protocol is not specific to `wg-manager`. But I agree we can be a bit more clear on the fact that the server should hold on to both peers, but only have one loaded into WireGuard at any single point. And that swapping between them is possible.

Reply:
Alright